src/Avalonia.Vulkan/Surfaces/X11VulkanPlatformSurface.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+using System.Runtime.InteropServices;
+using Avalonia.Platform;
+using Silk.NET.Vulkan;
+using Silk.NET.Vulkan.Extensions.KHR;
+
+namespace Avalonia.Vulkan.Surfaces
+{
+    internal class X11VulkanPlatformSurface : IVulkanPlatformSurface
+    {
+        private readonly IPlatformNativeSurfaceHandle _surfaceHandle;
+        private IntPtr _display = IntPtr.Zero;
+
+        [DllImport("libX11.so.6")]
+        public static extern IntPtr XOpenDisplay(IntPtr display);
```

Comment:
Optional: think about consuming these from a central place like XLib.cs as this DllImport now exists multiple times whereas the referenced binary (name) may change over time